File: src/Web/src/types/template.ts
export interface Template {
  id: string;
  name: string;
  description?: string;
  createdById: string;
  createdByName: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  stepCount: number;
}

export interface TemplateDetail {
  id: string;
  name: string;
  description?: string;
  createdById: string;
  createdByName: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  steps: TemplateStep[];
}

export interface TemplateStep {
  id: string;
  templateId: string;
  title: string;
  instructions?: string;
  sortOrder: number;
}

export interface CreateTemplateRequest {
  name: string;
  description?: string;
  steps: CreateTemplateStepRequest[];
}

export interface UpdateTemplateRequest {
  name?: string;
  description?: string;
  isActive?: boolean;
}

export interface CreateTemplateStepRequest {
  title: string;
  instructions?: string;
}

export interface UpdateTemplateStepRequest {
  title?: string;
  instructions?: string;
}

export interface ReorderStepsRequest {
  stepIds: string[];
}

File: src/Domain/Entities/Template.cs
namespace Domain.Entities;

public class Template
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public Guid CompanyId { get; set; }
    public Guid CreatedById { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Company Company { get; set; } = null!;
    public User CreatedBy { get; set; } = null!;
    public ICollection<TemplateStep> Steps { get; set; } = new List<TemplateStep>();
}
